aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -1700,7 +1700,7 @@ Tensor sparse_coo_to_sparse(const Tensor& self, const int64_t sparse_dim) {
       sparse_dim >= 0 && sparse_dim <= self.sparse_dim(), "sparse_dim argument for sparse_coo_to_sparse must be between 0 and number of sparse dimensions of original tensor");
   TORCH_CHECK(
      sparse_dim == self.sparse_dim(), "sparse_dim argument for sparse_coo_to_sparse must not be different than sparse dim of original tensor");
-  return self;
+  return self.requires_grad() ? self.clone() : self;
 }
```

Comment:
We'd better never rely on `requires_grad`. Better condition on `at::GradMode::is_enabled()`.

Reply:
The current usage of GradMode in aten suggests:
```suggestion
  return (at::GradMode::is_enabled() && self.requires_grad()) ? self.clone() : self;
}
```

Reply:
Added both checks.

Reply:
Closing the issue as out-of-date.